### PR TITLE
Add cloudsql build scripts.

### PR DESCRIPTION
**Pull request by dlorenc**

### CODE REVIEW

File: cloud_sql_proxy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM scratch
```
Comment (JustinBeckwith) [inline cloud_sql_proxy/Dockerfile]: scratch?

**Comment by dlorenc** [comment]
R: @andrewsg

File: cloud_sql_proxy/Dockerfile (hunk 1 above)
Comment (Carrotman42) [inline cloud_sql_proxy/Dockerfile]: The cloudsql proxy is a statically-built container. It doesn't need anything else (other than the root x509 certs for TLS connections, but those already reside on the VM are are mapped into the container).

File: cloud_sql_proxy/build.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+mkdir -p $GOPATH/src $GOPATH/pkg
+
+DEST=$GOPATH/src/github.com/GoogleCloudPlatform/cloudsql-proxy
+git clone https://github.com/GoogleCloudPlatform/cloudsql-proxy $DEST
+# Pin the version of the proxy. Bump this to get a new version.
+git -C $DEST checkout 1274cd3d89ac8826e1882355d60ffb2a0cdff116
+go get github.com/GoogleCloudPlatform/cloudsql-proxy/...
+echo "Building in $DEST"
+
+go build -o cloud_sql_proxy github.com/GoogleCloudPlatform/cloudsql-proxy/cmd/cloud_sql_proxy/
```
Comment (Carrotman42) [inline cloud_sql_proxy/build.sh]: You need a few more commands for the output binary to be in the correct format. This is what I used in the past:

CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o cloud_sql_proxy github.com/GoogleCloudPlatform/cloudsql-proxy/cmd/cloud_sql_proxy/

File: cloud_sql_proxy/Dockerfile (hunk 1 above)
Comment (dlorenc) [inline cloud_sql_proxy/Dockerfile]: Just added a second commit to make sure we statically build it :)
$ docker run f013
2016/01/11 23:59:23 No authentication method available! When not running on Google Compute Engine, provide the -credential_file flag.

File: cloud_sql_proxy/build.sh (hunk 2 above)
Comment (dlorenc) [inline cloud_sql_proxy/build.sh]: Should be there in the second commit. Thanks!

**Comment by andrewsg** [comment]
LGTM

**Pull request merged**